mypy/subtypes.py

```diff
@@ -144,7 +144,11 @@ def visit_callable_type(self, left: CallableType) -> bool:
             return all(is_subtype(left, item, self.check_type_parameter)
                        for item in right.items())
         elif isinstance(right, Instance):
+            if left.is_type_obj():
+                return is_subtype(left.ret_type, right)
```

Comment:
Mixing of a type object `C` with instances of the class `C`. These two lines can just be deleted. This is why other tests are failing, since the value `C`'s type was no longer a subtype of `type`, so `isinstance(..., C)` would fail.

Adding to the confusion, `messages.py` usually [prints the type of `C` as simply `C`](https://github.com/python/mypy/blob/master/mypy/messages.py#L164). Now that we have the `Type[C]` syntax, it would be clearer to use it in messages.